Tools/Scripts/libraries/webkitcorepy/webkitcorepy/subprocess_utils.py

```diff
@@ -44,12 +45,20 @@ def run(*popenargs, **kwargs):
 
         if difference:
             timeout = min(timeout or sys.maxsize, int(math.ceil(difference)))
-        if capture_output:
+        if capture_output or capture_and_stream_output:
             if ('stdout' in kwargs) or ('stderr' in kwargs):
-                raise ValueError('stdout and stderr arguments may not be used with capture_output.')
+                raise ValueError('stdout and stderr arguments may not be used with capture_output and capture_and_stream_output.')
             kwargs['stdout'] = subprocess.PIPE
-            kwargs['stderr'] = subprocess.PIPE
-        return subprocess.run(*popenargs, timeout=timeout, **kwargs)
+            kwargs['stderr'] = subprocess.STDOUT
+
+        output = []
+        process = subprocess.Popen(*popenargs, **kwargs)
+        if capture_and_stream_output:
+            for line in process.stdout:
+                sys.stdout.write(line)
+                output.append(line)
+        stdout, stderr = process.communicate(timeout=timeout)
+        return CompletedProcess(process.args, process.returncode, ''.join(output) or stdout, stderr)
```

Comment:
This is entirely too simplified compared with what the stdlib `subprocess.run` does, which various kills the process in some circumstances, raises `CalledProcessError` when appropriate, etc. etc.

Reply:
This is a really good point, thank you for this feedback. Upon reflection, I think I misinterpreted some feedback @JonWBedard gave me offline during development. My most recent commit makes this much more similar to how `subprocess.run` behaves.